geowave/src/main/scala/geotrellis/spark/io/geowave/GeowaveLayerWriter.scala

```diff
@@ -0,0 +1,253 @@
+import mil.nga.giat.geowave.core.geotime.ingest._
+import mil.nga.giat.geowave.core.store._
+import mil.nga.giat.geowave.core.store.adapter.statistics.StatsCompositionTool
+import mil.nga.giat.geowave.core.store.data.VisibilityWriter
+import mil.nga.giat.geowave.core.store.operations.remote.options.DataStorePluginOptions
+import mil.nga.giat.geowave.core.store.util.DataStoreUtils
+import mil.nga.giat.geowave.datastore.accumulo._
+import mil.nga.giat.geowave.datastore.accumulo.metadata._
+import mil.nga.giat.geowave.datastore.accumulo.operations.config.AccumuloOptions
+import mil.nga.giat.geowave.datastore.accumulo.util._
+import mil.nga.giat.geowave.datastore.accumulo.util.AccumuloUtils
+import org.apache.accumulo.core.client.security.tokens.PasswordToken
+import org.apache.accumulo.core.data.{ Key, Value }
+import org.apache.spark.rdd.RDD
+import org.apache.spark.SparkContext
+import org.geotools.coverage.grid.GridCoverage2D
+import org.geotools.coverage.processing.CoverageProcessor
+import org.geotools.factory.{ GeoTools, Hints }
+import org.opengis.coverage.grid.GridCoverage
+import org.opengis.parameter.ParameterValueGroup
+
+import java.io.{ DataInputStream, DataOutputStream, File }
+import java.util.UUID
+import javax.imageio.ImageIO
+import javax.media.jai.{ ImageLayout, JAI }
+import scala.collection.JavaConversions._
+import scala.collection.mutable.ListBuffer
+import scala.reflect._
+
+import resource._
+import spray.json._
+
+
+object GeowaveLayerWriter extends LazyLogging {
+
+  def write[
+    K: ClassTag,
+    V: ClassTag,
+    M: JsonFormat: GetComponent[?, Bounds[K]]
+  ](
+    coverageName: String,
+    rdd: RDD[(K, V)] with Metadata[M],
+    as: GeowaveAttributeStore,
+    accumuloWriter: AccumuloWriteStrategy
+  ): Unit = {
+    val metadata = implicitly[ClassTag[K]].toString match {
+      case "geotrellis.spark.SpatialKey" => rdd.metadata.asInstanceOf[TileLayerMetadata[SpatialKey]]
+      case t: String => throw new Exception(s"Unsupported Key Type: $t")
+    }
+
+    val crs = metadata.crs
+    val mt = metadata.mapTransform
+    val cellType = metadata.cellType.toString
+    val specimen = rdd.first
+
+    /* Construct (Multiband|)Tile to GridCoverage2D conversion function */
+    val geotrellisKvToGeotools: (((K, V)) => GridCoverage2D) = {
+      specimen match {
+        case (_: SpatialKey, _: Tile) => {
+          case (k: K, v: V) =>
+            val extent = mt(k.asInstanceOf[SpatialKey]).reproject(crs, LatLng)
+            val tile = v.asInstanceOf[Tile]
+            ProjectedRaster(Raster(tile, extent), LatLng).toGridCoverage2D
+        }
+        case (_: SpatialKey, _: MultibandTile) => {
+          case (k: K, v: V) =>
+            val extent = mt(k.asInstanceOf[SpatialKey]).reproject(crs, LatLng)
+            val tile = v.asInstanceOf[MultibandTile]
+            ProjectedRaster(Raster(tile, extent), LatLng).toGridCoverage2D
+        }
+      }
+    }
```

Comment:
This also will throw on anything but `Tile` and `MultibandTile`. If we need to restrict types, we'll have to do it via the type system